editor.planx.uk/src/pages/FlowEditor/components/Team/components/MembersTable.tsx

```diff
@@ -130,59 +163,68 @@ export const MembersTable = ({
                   />
                 </TableCell>
                 <TableCell>{member.email}</TableCell>
-                {showEditMemberButton && (
+                <TableCell>
                   <Permission.IsPlatformAdmin>
-                    <TableCell>
+                    {showEditMemberButton && (
                       <EditUserButton
                         onClick={() => {
-                          setShowUpdateModal(true);
-                          setInitialValues(member);
+                          editUser(member);
                         }}
-                        data-testId={`edit-button-${i}`}
+                        data-testid={`edit-button-${member.id}`}
                       >
                         Edit
                       </EditUserButton>
-                    </TableCell>
+                    )}
+                  </Permission.IsPlatformAdmin>
+                </TableCell>
+                <TableCell>
+                  <Permission.IsPlatformAdmin>
```

Comment:
Permission level moved inside `<TableCell>` to ensure consistent UI across different permission levels